lib/Db/MessageMapper.php

```diff
@@ -912,7 +914,7 @@ public function findNewIds(Mailbox $mailbox, array $ids): array {
 		$sub = $this->db->getQueryBuilder();
 
 		$subSelect = $sub
-			->select($sub->func()->max('uid'))
+			->select($sub->func()->max('sent_at'))
```

Comment:
Note to myself: A message moved from A to B get's a new uid.

Delete the last message from a thread and delete the second last message afterwards. In the trash the second last appears before the last because the uid is higher.